src/core/utils.js

```diff
@@ -949,3 +898,54 @@ export class CaseInsensitiveMap extends Map {
     return super.delete(key.toLowerCase());
   }
 }
+
+export class Err extends Error {
```

Comment:
Consider other names: 
- `ReSpecError`: too long
- `RsError`?

Reply:
RsError would be better IMO